File: frontend/src/services/especialidadService.js
import api from './api';

export const especialidadService = {
  getAll: async () => {
    const response = await api.get('/especialidades');
    return response.data;
  },

  getById: async (id) => {
    const response = await api.get(`/especialidades/${id}`);
    return response.data;
  },

  create: async (especialidad) => {
    const response = await api.post('/especialidades', especialidad);
    return response.data;
  },

  update: async (id, especialidad) => {
    const response = await api.put(`/especialidades/${id}`, especialidad);
    return response.data;
  },

  delete: async (id) => {
    const response = await api.delete(`/especialidades/${id}`);
    return response.data;
  },

  buscar: async (termino) => {
    const response = await api.get(`/especialidades/buscar/${termino}`);
    return response.data;
  }
};

File: src/main/java/com/tecsup/examen2/Medicos/service/EspecialidadService.java
package com.tecsup.examen2.Medicos.service;

import org.springframework.stereotype.Service;
import com.tecsup.examen2.Medicos.repository.EspecialidadRepository;
import com.tecsup.examen2.Medicos.model.Especialidad;
import java.util.*;

@Service
public class EspecialidadService {

    private final EspecialidadRepository especialidadRepo;

    public EspecialidadService(EspecialidadRepository especialidadRepo) {
        this.especialidadRepo = especialidadRepo;
    }

    public List<Especialidad> findAll() {
        return especialidadRepo.findAll();
    }

    public Optional<Especialidad> findById(String id) {
        return especialidadRepo.findById(id);
    }

    public Especialidad create(Especialidad especialidad) {
        return especialidadRepo.save(especialidad);
    }

    public Especialidad update(String id, Especialidad updated) {
        return especialidadRepo.findById(id).map(existing -> {
            existing.setNombre(updated.getNombre());
            existing.setDescripcion(updated.getDescripcion());
            return especialidadRepo.save(existing);
        }).orElseThrow(() -> new RuntimeException("Especialidad no encontrada"));
    }

    public void delete(String id) {
        especialidadRepo.deleteById(id);
    }

    public List<Especialidad> buscar(String termino) {
        return especialidadRepo.findByNombreContainingIgnoreCase(termino);
    }
}
